test/passing/comments.ml

```diff
@@ -129,3 +129,31 @@ type t = {
 a : int [@default a]
 (* comment *)
 ;  b : flag }
+
+let foo =
+  (* foo *)
+  match (* comment about entire pattern matching *)
+    e with k -> z
```

Comment:
Sorry for the piecemeal commentary, but what about the case where the comment and match are on the same line?
```
(* foo *) match (* bar *) e with...
```